chain-core/src/tx/data/attribute.rs

```diff
@@ -9,11 +12,62 @@ use crate::tx::data::access::TxAccessPolicy;
 #[derive(Debug, Default, PartialEq, Eq, Clone, Encode, Decode)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct TxAttributes {
+    #[cfg_attr(feature = "serde", serde(serialize_with = "serialize_chain_hex_id"))]
+    #[cfg_attr(
+        feature = "serde",
+        serde(deserialize_with = "deserialize_chain_hex_id")
+    )]
     pub chain_hex_id: u8,
     pub allowed_view: Vec<TxAccessPolicy>,
     // TODO: other attributes, e.g. versioning info
 }
 
+#[cfg(feature = "serde")]
+fn serialize_chain_hex_id<S>(
+    chain_hex_id: &u8,
```

Comment:
Do the custom serialisers need it by reference?

There's a clippy warning:
```
error: this argument (1 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)
3841  --> chain-core/src/tx/data/attribute.rs:27:19
3842   |
384327 |     chain_hex_id: &u8,
3844   |                   ^^^ help: consider passing by value instead: `u8`
3845   |
3846   = note: `-D clippy::trivially-copy-pass-by-ref` implied by `-D warnings`
3847   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref
3848
3849error: aborting due to previous error
```

If it's required by the serialiser / it's not possible to pass by value, then you can add an annotation for clippy to ignore the `clippy::trivially-copy-pass-by-ref` warning in this function

Reply:
Yes, yesterday I tried to use pass by value and it seems it does not work. I will see if it is required and ignore the warning if needed.